data/.schema.yaml

```diff
@@ -144,15 +145,25 @@ properties:
             additionalProperties: false
             properties:
               time:
-                type: number
                 description: >
                   Temporal offset for the sample, e.g., relative to symptom onset, drug
                   ingestion, or hospital admission. The reference event should be
                   defined in the dataset description.
+                oneOf:
+                  - type: number
+                  - const: unknown
               value:
                 $ref: "#/$defs/negative_positive_number"
               analyte:
+                description: >
+                  Analyte for which a measurement is reported. The analyte key must
+                  appear in the dictionary of `analytes` metadata.
                 type: string
+              sample_id:
```

Comment:
Is this going to be values from raw data or auto-generated?

Reply:
I think this would only ever be from raw data because we'd need the raw data to tell us if measurement A and measurement B were performed on the same sample.

Reply:
Sounds good, let's discuss on Friday. In this case, there are different samples taken on the same day (see below).

```python
import yaml
import pandas as pd


with open("../data/natarajan2022.yaml") as fp:
    data = yaml.safe_load(fp)

for i, participant in enumerate(data["participants"]):
    measurements = pd.DataFrame(participant["measurements"])
    n_unique_samples_by_day = measurements.groupby("time").sample_id.nunique()
    print(f"participant {i}: {n_unique_samples_by_day[n_unique_samples_by_day > 1].to_dict()}")
```

This yields the following output. E.g., for participant 0, there are two distinct samples on days 142 and 148.

```
participant 0: {142: 2, 182: 2}
participant 1: {135: 2}
participant 2: {147: 2, 253: 2}
participant 3: {31: 2, 180: 2, 201: 2}
participant 4: {}
participant 5: {14: 2}
participant 6: {123: 2}
participant 7: {3: 2, 131: 2}
participant 8: {5: 2}
participant 9: {4: 2}
participant 10: {3: 2, 141: 2, 177: 2}
participant 11: {1: 2}
participant 12: {}
participant 13: {}
participant 14: {'unknown': 2}
participant 15: {150: 2, 205: 2, 'unknown': 2}
participant 16: {}
participant 17: {14: 2, 28: 2, 124: 2, 197: 2}
participant 18: {5: 2, 195: 2}
participant 19: {113: 2}
participant 20: {}
participant 21: {14: 2, 29: 2}
participant 22: {210: 2}
participant 23: {210: 2}
participant 24: {14: 4, 27: 2}
participant 25: {14: 2, 29: 2}
participant 26: {29: 2}
participant 27: {14: 2, 29: 2}
participant 28: {28: 2, 123: 2, 195: 2}
participant 29: {13: 2, 28: 2, 138: 2, 200: 2}
participant 30: {}
participant 31: {10: 2, 26: 2, 116: 2}
participant 32: {10: 2, 27: 2, 151: 2, 227: 2}
participant 33: {10: 2, 27: 2, 151: 2, 227: 2}
participant 34: {3: 2, 14: 2, 'unknown': 2}
participant 35: {4: 2, 13: 2, 30: 2, 130: 2}
participant 36: {28: 2, 'unknown': 2}
participant 37: {2: 2, 13: 2}
participant 38: {}
participant 39: {4: 2, 14: 2, 'unknown': 2}
participant 40: {14: 2, 100: 2, 'unknown': 2}
participant 41: {5: 2, 15: 2}
participant 42: {0: 2, 10: 2, 28: 2, 120: 2, 214: 2}
participant 43: {14: 2, 28: 2}
participant 44: {6: 2, 30: 2}
participant 45: {211: 2, 'unknown': 6}
participant 46: {211: 2}
participant 47: {27: 2, 103: 2, 189: 2, 'unknown': 4}
participant 48: {}
participant 49: {3: 2, 14: 2, 28: 2, 109: 2, 210: 2}
participant 50: {1: 2, 28: 2, 'unknown': 2}
participant 51: {4: 2, 'unknown': 4}
participant 52: {1: 2, 15: 2}
participant 53: {1: 2, 8: 2, 28: 2, 107: 2}
participant 54: {1: 2, 9: 2, 20: 2, 26: 2}
participant 55: {}
participant 56: {1: 2, 103: 2, 'unknown': 2}
participant 57: {3: 2}
participant 58: {3: 2, 10: 2, 25: 2, 103: 2, 202: 2}
participant 59: {0: 2}
participant 60: {1: 2, 28: 2, 'unknown': 2}
participant 61: {1: 2, 15: 2}
participant 62: {1: 2, 12: 2, 'unknown': 2}
participant 63: {'unknown': 2}
participant 64: {3: 2, 14: 2, 109: 2, 196: 2}
participant 65: {14: 2, 104: 2, 'unknown': 2}
participant 66: {14: 2, 27: 2}
participant 67: {14: 2, 23: 2}
participant 68: {2: 2, 4: 2, 116: 2}
participant 69: {11: 2, 'unknown': 2}
participant 70: {}
participant 71: {117: 2, 'unknown': 4}
participant 72: {110: 2, 'unknown': 4}
participant 73: {116: 2, 'unknown': 5}
participant 74: {112: 2, 192: 2}
participant 75: {'unknown': 3}
participant 76: {}
participant 77: {'unknown': 4}
participant 78: {5: 2, 116: 2}
participant 79: {3: 2, 21: 2, 28: 2, 114: 2}
participant 80: {0: 2, 28: 2, 97: 2, 'unknown': 2}
participant 81: {0: 2, 5: 2, 128: 2}
participant 82: {2: 2, 28: 2, 115: 2, 'unknown': 4}
participant 83: {10: 2, 28: 2, 103: 2, 'unknown': 2}
participant 84: {27: 2, 'unknown': 2}
participant 85: {3: 2, 28: 2, 112: 2, 'unknown': 2}
participant 86: {28: 2, 'unknown': 3}
participant 87: {3: 2, 22: 2, 28: 2}
participant 88: {2: 2, 'unknown': 2}
participant 89: {15: 2, 27: 2}
participant 90: {3: 2, 102: 2, 'unknown': 4}
participant 91: {31: 2, 'unknown': 4}
participant 92: {'unknown': 3}
participant 93: {0: 2, 112: 2}
participant 94: {}
participant 95: {'unknown': 2}
participant 96: {102: 2, 'unknown': 2}
participant 97: {2: 2, 27: 2, 112: 2}
participant 98: {2: 2, 119: 2}
participant 99: {111: 2, 'unknown': 4}
participant 100: {3: 2, 120: 2, 'unknown': 4}
participant 101: {'unknown': 4}
participant 102: {111: 2, 'unknown': 3}
participant 103: {106: 2}
participant 104: {116: 2, 'unknown': 2}
participant 105: {}
participant 106: {'unknown': 3}
participant 107: {139: 2}
participant 108: {155: 2}
participant 109: {3: 2, 'unknown': 2}
participant 110: {'unknown': 3}
participant 111: {55: 2, 'unknown': 2}
participant 112: {2: 2, 120: 2, 'unknown': 3}
```